sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/TrackNotImplementedException.cs

```diff
@@ -56,6 +64,20 @@ protected override void Initialize(SonarAnalysisContext context)
                     }
                 },
                 SyntaxKind.ThrowStatement);
```

Comment:
you can add here a second SyntaxKind which is `SyntaxKindEx.ThrowExpression` , and have a method to extract what you need and then avoid duplication for checking the type against the KnownType